srtcore/core.cpp

```diff
@@ -9525,65 +9525,75 @@ bool srt::CUDT::packUniqueData(CPacket& w_packet, time_point& w_origintime)
+                // the sending buffer yet, and therefore we are in a situation as just
+                // after connection. No packets in the buffer, no packets are sent,
+                // no ACK to be awaited. We can screw up all the variables that are
+                // initialized from ISN just after connection.
+                LOGC(qslog.Warn,
+                     log << CONID() << "packData: Fixing EXTRACTION sequence " << m_iSndCurrSeqNo
+                         << " from SCHEDULING sequence " << w_packet.m_iSeqNo << " for the first packet: DIFF="
+                         << packetspan << " STAMP=" << BufferStamp(w_packet.m_pcData, w_packet.getLength()));
+
+                // Additionally send the drop request to the peer so that it
+                // won't stupidly request the packets to be retransmitted.
+                // Don't do it if the difference isn't positive or exceeds the threshold.
+                if (packetspan > 0)
+                {
+                    int32_t seqpair[2];
+                    seqpair[0] = m_iSndCurrSeqNo;
+                    seqpair[1] = CSeqNo::decseq(w_packet.m_iSeqNo);
+                    const int32_t no_msgno = 0;
+                    LOGC(qslog.Debug,
+                         log << "packData: Sending DROPREQ (ISN FIX): SEQ: " << seqpair[0] << " - " << seqpair[1]
+                             << " (" << packetspan << " packets)");
+                    sendCtrl(UMSG_DROPREQ, &no_msgno, seqpair, sizeof(seqpair));
+                    // In case when this message is lost, the peer will still get the
+                    // UMSG_DROPREQ message when the agent realizes that the requested
+                    // packet are not present in the buffer (preadte the send buffer).
+                }
 
-            // In case when this message is lost, the peer will still get the
-            // UMSG_DROPREQ message when the agent realizes that the requested
-            // packet are not present in the buffer (preadte the send buffer).
+                // Override extraction sequence with scheduling sequence.
+                m_iSndCurrSeqNo = w_packet.m_iSeqNo;
+                ScopedLock ackguard(m_RecvAckLock);
+                m_iSndLastAck = w_packet.m_iSeqNo;
+                m_iSndLastDataAck = w_packet.m_iSeqNo;
+                m_iSndLastFullAck = w_packet.m_iSeqNo;
+                m_iSndLastAck2 = w_packet.m_iSeqNo;
+            }
+            else
+            {
+                // There will be a serious data discrepancy between the agent and the peer.
+                LOGC(qslog.Error,
+                     log << CONID() << "IPE: packData: Fixing EXTRACTION sequence " << m_iSndCurrSeqNo
+                         << " from SCHEDULING sequence " << w_packet.m_iSeqNo << " in the middle of transition: DIFF="
+                         << packetspan << " STAMP=" << BufferStamp(w_packet.m_pcData, w_packet.getLength()));
```

Comment:
IPE and log message about fixing the sequence, without actually fixing it?

It seems like the general approach to the sync up here should be:
- sync up if `w_packet.m_iSeqNo` is ahead of the `m_iSndCurrSeqNo`;
   - IPE if not ISN, although sending can probably still go further without serious issues besides of dropping those initial packets.
- IPE if it is behind, and probably don't send this packet.
 
In the latter case of IPE, it looks reasonable to change the socket state to broken. As it is a highly unexpected situation, I am not sure if proceeding further would not turn into bigger trouble.